lib/sparse/src/index/search_context.rs

```diff
@@ -191,6 +203,96 @@ impl<'a> SearchContext<'a> {
         }
     }
 
+    /// Intersects remaining posting lists to compute scores in a single pass.
+    ///
+    /// Expects at least two posting lists.
+    /// Allocates `records_count` intermediate scores for the intersections.
+    fn fast_eval_posting_lists<F: Fn(PointOffsetType) -> bool>(
+        &mut self,
+        records_count: usize,
+        min_id: u32,
+        filter_condition: &F,
+    ) {
+        let posting_len = self.postings_iterators.len();
+        debug_assert!(posting_len >= 2);
+        debug_assert!(records_count < FAST_EVAL_THRESHOLD);
+        // shortest lists first to allocate less intermediate scores
+        self.postings_iterators.sort_unstable_by(|a, b| {
+            a.posting_list_iterator
+                .len_to_end()
+                .cmp(&b.posting_list_iterator.len_to_end())
+        });
+        // save intermediate score starting from `min_id` at index 0
+        let mut intermediate_scores: Vec<Option<ScoreType>> = vec![None; records_count];
```

Comment:
my initial implementation was saving scores in a `Hashmap + Ahash` but a Vector is significantly faster